src/language_features/lean.rs

```diff
@@ -77,3 +77,76 @@ pub fn plain_goal(meta: EditorMeta, params: EditorPlainGoalParams, ctx: &mut Con
         },
     );
 }
+
+#[derive(Debug, Clone)]
+pub struct EditorPlainTermGoalParams {
+    pub position: KakounePosition,
+    pub buffer: String,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct PlainTermGoalParams {
+    pub text_document: TextDocumentIdentifier,
+    pub position: Position,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct PlainTermGoalResponse {
+    pub goal: String,
+    pub range: Range,
+}
+
+pub struct PlainTermGoalRequest {}
+
+impl Request for PlainTermGoalRequest {
+    type Params = PlainTermGoalParams;
+    type Result = Option<PlainTermGoalResponse>;
```

Comment:
I think we should reuse `PlainGoalParams` as-is here rather than defining our own type,
that might be unconventional but I think it would help with code sharing